src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -184,13 +179,56 @@ void GcsAutoscalerStateManager::GetClusterResourceConstraints(
   }
 }
 
+void GcsAutoscalerStateManager::UpdateResourceLoadAndUsage(
+    const rpc::ResourcesData &data) {
+  NodeID node_id = NodeID::FromBinary(data.node_id());
+  auto iter = node_resource_info_.find(node_id);
+  if (iter == node_resource_info_.end()) {
+    iter = node_resource_info_
+               .emplace(node_id, std::make_pair(absl::Now(), rpc::ResourcesData()))
+               .first;
+  }
```

Comment:
gcs node manager should be the SOT of whether a node is part of the cluster or not: we should just ignore the message if the node is not in gcs node manager.

Reply:
should we just initialize the data with `OnNodeAdd`?

Reply:
Yeah